docs/ruleset_functions/get_number_of_floors.md

```diff
@@ -0,0 +1,54 @@
+# get_number_of_floors
+**Schema Version:** 0.0.23  
+
+**Description:** gets the number of floors in the building - logic is based on looking at the elevation of zones - zones with an elevation of <0 are considered underground and are not counted.  Parking Garages are not counted
+
+**Inputs:**
+- **zone_input_list** - the list of the zones for which we want to get the number of floors (changed from RMI to this so that we can get the number of floors for a limited list of zones, for example for the function get_hvac_systems_5_6_serving_multiple_floors_b)
+
+**Returns:**  
+- **result**: an integer indicating the number of floors
+ 
+**Function Call:**
+
+## Logic:
+- make a list of eligible zones in the building: `zone_list = []`
+	- loop through each zone in the input list: `for zone in zone_input_list: `
+		- eligible zones are zones with an elevation greater than zero: `if zone.elevation > 0`
+			- loop throught the spaces in the zone: `for space in zone:`
+				- eligible zones are zones that are not parking garage zones if any of the spaces are not PARKING_GARAGE, then this zone is not exclusively a parking garage space: `if space.lighting_space_type != PARKING_GARAGE:`
+					- add the zone to the zone_list: `zone_list.append(zone)`
+					- break out of the loop: `break`
+
+- sort the list by zone elevation: `zone_list.sort(key = lambda x: x.elevation)`
+
+- create an integer num_of_floors: `num_of_floors = 0`
+
+- when we find floors, we'll remove the zones from the zone_list, until there are no more zones in the list, so we create a loop using `while` and the condition `len(zone_list) > 0`: `while len(zone_list) > 0:`
+	- the first zone in the list will be the one with the lowest elevation, create variable elevation equal to this first zone's elevation: `elevation = zone_list[0].elevation`
+	- until we find a zone with an elevation greater than 7.5' higher than this base elevation, count the zones as being on the same floor - this step insures that zones that have a slightly different elevation, such as a single step up or down do not get counted as being on different floors: `while zone_list[0].elevation <= elevation + 7.5:`
```

Comment:
I am not sure why we are using 7.5' in the calculation. Based on the definition of "story" in 90.1 copied below, wouldn't the number of floors in the building be equal to the number of unique "elevations"?  I do not think we should worry about plenums because it is my understanding that they should be combined with the adjacent spaces in the RMD and not listed as a separate zone - see note to adjacent_to field in the schema/

story: portion of a building that is between one finished floor level and the next higher finished floor level or
the roof, provided, however, that a basement or cellar shall not be considered a story.

Reply:
if we are counting below-grade floors, and we the user / UI to name the floors correctly, we can use floor_name and not bother with elevation

(I was using 7.5' to capture spaces that have a slightly different elevation, such as a renovation where there are 1-2 steps up or down to the new part of the building - these zones would have a different elevation, but should still count as the same floor)

Reply:
as discussed on RDS call, we will use zone elevation instead of the zone name